File: rn-expo/src/domain/MovieRepository.ts
import { MovieDetail, MoviePage } from "./entities";

export type Result<T> =
  | { success: true; data: T }
  | { success: false; error: string };

export interface MovieRepository {
  getPopularMovies(page: number): Promise<Result<MoviePage>>;
  getMovieDetail(id: number): Promise<Result<MovieDetail>>;
}

File: kmp-native/shared/src/commonMain/kotlin/com/stackbenchmark/kmpnative/domain/repository/MovieRepository.kt
package com.stackbenchmark.kmpnative.domain.repository

import com.stackbenchmark.kmpnative.domain.entity.MovieDetail
import com.stackbenchmark.kmpnative.domain.entity.MoviePage

interface MovieRepository {
    suspend fun getPopularMovies(page: Int): Result<MoviePage>
    suspend fun getMovieDetail(id: Int): Result<MovieDetail>
}
